src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs

```diff
@@ -452,6 +458,25 @@ internal static object GetTaskResult(Task task)
             return null;
         }
 
+        private void WarnIfDeprecatedNugetReferenceFound(Diagnostic diagnostic)
+        {
+            string functionDirectory = Metadata.FunctionDirectory;
+            string deprecatedProjectPath = Path.Combine(functionDirectory, DotNetConstants.DeprecatedProjectFileName);
+            string extensionsProjectPath = Path.Combine(functionDirectory, ScriptConstants.ExtensionsProjectFileName);
+            const string warningString = "You may be referencing NuGet packages incorrectly. The file '{0}' should not be used to reference NuGet packages. Try creating a '{1}' file instead. Learn more: https://go.microsoft.com/fwlink/?linkid=2091419";
+
+            if (File.Exists(deprecatedProjectPath))
+            {
+                string warning = string.Format(warningString, deprecatedProjectPath, DotNetConstants.ProjectFileName);
+                FunctionLogger.LogWarning(warning);
+            }
+            if (File.Exists(extensionsProjectPath))
+            {
+                string warning = string.Format(warningString, extensionsProjectPath, DotNetConstants.ProjectFileName);
+                FunctionLogger.LogWarning(warning);
+            }
```

Comment:
What if none of these files (nor the function.proj) are found? Can we write out something with the link to the docs in that case, too?